test/e2e/smoketest.go

```diff
@@ -58,10 +58,11 @@ func productionSmokeTest(jaegerInstanceName, smokeTestNamespace string) {
 	queryPodPrefix := jaegerInstanceName + "-query"
 	collectorPodPrefix := jaegerInstanceName + "-collector"
 
+	// Use ingress for k8s or on OpenShift if we have an insecure route
 	var apiTracesEndpoint string
-	route, hasInsecureRoute := getInsecureRoute(jaegerInstanceName)
-	if hasInsecureRoute {
-		apiTracesEndpoint = fmt.Sprintf("https://%s/api/traces", route.Spec.Host)
+	hasInsecureRoute := hasInsecureRoute(jaegerInstanceName)
+	if !isOpenShift(t) || hasInsecureRoute {
```

Comment:
Would be better to move the test of openshift into the `hasInsecureRoute` function, and then this condition could just be `if hasInsecureRoute {`.

Reply:
Also, then maybe change the method name, so doesn't imply specific to route - e.g. `hasInsecureEndpoint`?